lib/workers/repository/init/vulnerability.ts

```diff
@@ -171,72 +135,90 @@ export async function detectVulnerabilityAlerts(
+                heading = `[${heading}](${advisory.references[0].url})`;
+              }
+              content += heading;
+              content += '\n\n';
 
-                content += sanitizeMarkdown(description);
-                return content;
-              }),
-            );
-          } catch (err) /* istanbul ignore next */ {
-            logger.warn({ err }, 'Error generating vulnerability PR notes');
-          }
-          // TODO: types (#22198)
-          const allowedVersions =
-            datasource === PypiDatasource.id
-              ? `==${val.firstPatchedVersion!}`
-              : val.firstPatchedVersion;
-          const matchFileNames =
-            datasource === GoDatasource.id
-              ? [fileName.replace('go.sum', 'go.mod')]
-              : [fileName];
-          let matchRule: PackageRule = {
-            matchDatasources: [datasource],
-            matchPackageNames: [depName],
-            matchCurrentVersion,
-            matchFileNames,
-          };
-          if (
-            datasource === PypiDatasource.id &&
-            normalizePythonDepName(depName) !== depName
-          ) {
-            matchRule.matchPackageNames?.push(normalizePythonDepName(depName));
+              content += sanitizeMarkdown(description);
+              return content;
+            }),
+          );
+        } catch (err) /* istanbul ignore next */ {
+          logger.warn({ err }, 'Error generating vulnerability PR notes');
+        }
+
+        const allowedVersions =
+          datasource === PypiDatasource.id
+            ? `==${val.firstPatchedVersion}`
+            : val.firstPatchedVersion;
+        const matchFileNames =
+          datasource === GoDatasource.id
+            ? [fileName.replace('go.sum', 'go.mod')]
+            : [fileName];
+        let matchCurrentVersion = '';
+        if (!matchCurrentVersion.length) {
+          if (datasource === MavenDatasource.id) {
+            matchCurrentVersion = `(,${val.firstPatchedVersion})`;
+          } else {
+            matchCurrentVersion = `< ${val.firstPatchedVersion}`;
           }
-          // Remediate only direct dependencies
-          matchRule = {
-            ...matchRule,
-            allowedVersions,
-            prBodyNotes,
-            isVulnerabilityAlert: true,
-            force: {
-              ...config.vulnerabilityAlerts,
-            },
-          };
-          alertPackageRules.push(matchRule);
         }
+        if (datasource === PypiDatasource.id) {
+          matchCurrentVersion = matchCurrentVersion.replace(
+            regEx(/^= /),
+            '== ',
+          );
+        }
+        if (
+          datasource === GithubTagsDatasource.id ||
+          datasource === MavenDatasource.id
+        ) {
+          // GitHub Actions uses docker versioning, which doesn't support `= 1.2.3` matching, so we strip the equals
+          matchCurrentVersion = matchCurrentVersion.replace(/^=\s*/, '');
+        }
```

Comment:
can we use a switch case here for datasource? it seems the code becomes more clean then?